File: C++/55. Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int step = 0, i;
        for(i=0;i<nums.size() && i<=step;i++)
            step = max(i+nums[i],step);
        return i == nums.size();
    }
};

File: Python/55. Jump Game.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        i, step = 0, 0
        while i < len(nums):
            if i > step:
                return False
            step = max(i+nums[i], step)
            i += 1
        return i == len(nums)